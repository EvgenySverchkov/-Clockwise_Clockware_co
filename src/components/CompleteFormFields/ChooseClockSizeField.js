import React from "react";

import RadioBtn from "../FormComponents/RadioBtn";
import Label from "../FormComponents/Label";
import FormGroup from "../FormComponents/FormGroup";

const ChooseClockSizeField = ({ changeHandler }) => (
  <FormGroup isRow={false}>
    <div className="mb-2 font-weight-bold">Choose size of clock</div>
    <RadioBtn
      id={"smallSize"}
      value={"small"}
      name={"size"}
      chngHandler={changeHandler}
    >
      <Label forId={"smallSize"} isFontWeight={false}>
        Small
      </Label>
    </RadioBtn>
    <RadioBtn
      id={"middleSize"}
      value={"middle"}
      name={"size"}
      chngHandler={changeHandler}
    >
      <Label forId={"middleSize"} isFontWeight={false}>
        Middle
      </Label>
    </RadioBtn>
    <RadioBtn
      id={"largeSize"}
      value={"large"}
      name={"size"}
      chngHandler={changeHandler}
    >
      <Label forId={"largeSize"} isFontWeight={false}>
        Large
      </Label>
    </RadioBtn>
  </FormGroup>
);

export default ChooseClockSizeField;
