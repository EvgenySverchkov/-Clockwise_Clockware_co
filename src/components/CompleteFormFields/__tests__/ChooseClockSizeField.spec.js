import React from "react";

import {create} from "react-test-renderer";
import ChooseClockSizeField from "../ChooseClockSizeField";

it("<ChooseClockSizeField /> test which is used in forms", ()=>{
    const mockProps = {
        changeHandler: jest.fn()
    }
    const component = create(<ChooseClockSizeField {...mockProps}/>);
    expect(component.toJSON()).toMatchSnapshot();
})