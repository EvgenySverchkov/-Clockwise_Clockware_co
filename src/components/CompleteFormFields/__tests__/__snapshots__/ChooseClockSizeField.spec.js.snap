// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChooseClockSizeField /> test which is used in forms 1`] = `
<div
  className="form-group  text-center text-sm-left col-12 col-md-10 col-lg-8"
>
  <div
    className="mb-2 font-weight-bold"
  >
    Choose size of clock
  </div>
  <div
    className="form-check form-check-inline"
  >
    <input
      className="form-check-input"
      id="smallSize"
      name="size"
      onChange={[MockFunction]}
      type="radio"
      value="small"
    />
    <label
      className="pl-0 pr-0 col-sm-4 col-md-4 col-form-label "
      htmlFor="smallSize"
    >
      Small
    </label>
  </div>
  <div
    className="form-check form-check-inline"
  >
    <input
      className="form-check-input"
      id="middleSize"
      name="size"
      onChange={[MockFunction]}
      type="radio"
      value="middle"
    />
    <label
      className="pl-0 pr-0 col-sm-4 col-md-4 col-form-label "
      htmlFor="middleSize"
    >
      Middle
    </label>
  </div>
  <div
    className="form-check form-check-inline"
  >
    <input
      className="form-check-input"
      id="largeSize"
      name="size"
      onChange={[MockFunction]}
      type="radio"
      value="large"
    />
    <label
      className="pl-0 pr-0 col-sm-4 col-md-4 col-form-label "
      htmlFor="largeSize"
    >
      Large
    </label>
  </div>
</div>
`;
